lib/app.rb

```diff
@@ -153,4 +150,20 @@ def run
       command = @input_stream.gets.chomp
     end
   end
+
+  private
+
+  def switch_current_project
+    output_stream.puts("\e[0;35mEnter a project name:\e[0m")
+    get_user_input
+    set_current_project(last_input)
+  end
+
+  def set_current_project(name)
+    @current_project = projects.detect { |p| p.keys.first == name }
+  end
+
+  def get_user_input
+    @last_input = input_stream.gets.chomp
```

Comment:
This feels like a hack to support the refactors living in methods instead of moving into their own classes - I only did this so the error message for not finding the project could be printed.

Reply:
I think things are going to be a bit hacky for a bit. You can't refactor the whole world at once, so this step is about finding the right boundary for that work.

I think `@last_input` and the related accessor is not an intuitive name. In the stories, we refer to these inputs  as commands. So something in that ilk would be better for product alignment ... and we do have a product strangely enough